build_image.py

```diff
@@ -24,59 +25,112 @@
+                    1. args.gittoken: The GITHUB_TOKEN secret
+                    2. args.tenant: Carol app's tenant
+                    3. args.appname: Carol app's organization
+                    4. args.org: Carol app's name
+                    5. args.apikey: Carol app's connector id
+                    6. args.connectorid: Carol app's API key
+                    7. args.manifestpath: Manifest.json file path.
 
-r = app.edit_manifest(app_name=app_name, manifest=manifest)
+            Carol app name. It will overwrite the app name used in Carol() initialization.
+        tasks: `list` default `carolApp`
+            Possible values `carolApp` or `carolConnect`
+        Returns:
+    """
 
-tasks = app.build_docker_git(git_token=git_token)
+    git_token = args.gittoken
+    tenant = args.tenant
+    app_name = args.appname
+    organization = args.org
+    api_key = args.apikey
+    connector_id = args.connectorid
+    manifest_path = args.manifestpath
 
-carol_task = Tasks(login)
+    login = Carol(
+        domain=tenant,
+        app_name=app_name,
+        organization=organization,
+        auth=ApiKeyAuth(api_key),
+        connector_id=connector_id
+    )
 
-indices = {task['mdmId']:0 for task in tasks}
+    manifest_path = os.path.join(manifest_path, 'manifest.json')
 
-while all([carol_task.get_task(task['mdmId']).task_status in ['READY', 'RUNNING'] for task in tasks]):
+    with open(manifest_path) as f:
+        manifest = json.load(f)
 
-    to_remove = []
+    logger.debug(f'Manifest: {manifest}')
 
-    for task in tasks[:]:
-        task_id = task['mdmId']
+    filters = ['''labels."k8s-pod/kind"="build"''', f'''labels."k8s-pod/appName":"{app_name}"''']
+
+    if tasks is None:
+        tasks = create_build_jobs(login, manifest, app_name, git_token)
+        tasks = [task['mdmId'] for task in tasks]
+        utc_now = datetime.utcnow().isoformat(timespec='milliseconds') + 'Z'
+        filters.append(f'''timestamp>="{utc_now}"''')
+
+    carol_task = Tasks(login)
+    indices = {task: 0 for task in tasks}
+
+    c = Compute(login)
+    r = c.get_app_logs(filters=filters, page_size=100)
+
+    fail = False  # for exit code purpose
+    while (all([carol_task.get_task(task).task_status in ['READY', 'RUNNING'] for task in tasks]) and len(tasks) > 1) \
+            or len(r['logEntries']) > 0:
+
+        to_remove = []
+
+        r = c.get_app_logs(filters=filters, page_size=100)
+
+        if r['logEntries']:
+            for entry in r['logEntries']:
+                logger.debug(entry['payload']['data'])
+            utc_now = datetime.utcfromtimestamp(entry['timestamp'] / 1000.0).isoformat(timespec='milliseconds') + 'Z'
+            filters = ['''labels."k8s-pod/kind"="build"''', f'''labels."k8s-pod/appName":"{app_name}"''',
+                       f'''timestamp>="{utc_now}"''']
+
+        time.sleep(1)
+
+    for task_id in tasks:
         logs = carol_task.get_logs(task_id)
 
         if len(logs) > indices[task_id]:
```

Comment:
Since this iteration is not inside the while loop anymore we can iterate `logs` directly instead of using its indices.

Reply:
fixed